Feature/thumbnails

mgdaily opened the pull request:
These are the changes needed to support the ingester and science archive in storing thumbnails. Here we just add some validation to make sure some additional thumbnail metadata is included in the payload. Also add a set of thumbnail filetypes which the ingester uses to know where to send the file.

Commit:
Add basic thumbnail support.

This adds an environment config for thumbnail filetypes, so we know which files to ship and treat as thumbnails in the science archive.

Commit:
Update README with THUMBNAIL_FILETYPES config

Commit:
WIP for thumbnails

Commit:
MMake clear that we need frame basename if w'ere ingesting a thumbnail

Commit:
Make environment variable match setting name

Commit:
Fix up README for settings

Commit:
Fix up lockfile

Commit:
Add ThumbnailFile class

Remove mapping in settings, update filestore path, etc...

Commit:
Merge branch 'feature/thumbnails' of github.com:observatorycontrolsystem/ocs_archive into feature/thumbnails

Commit:
Pull the header validation for thumbnails out

Also add a couple of tests for thumbnail files

jnation3406 commented on ocs_archive/settings/settings.py:
```diff
@@ -44,6 +44,9 @@ def get_tuple_from_environment(variable_name, default):
 REQUIRED_HEADERS = get_tuple_from_environment('REQUIRED_HEADERS', 'PROPID,DATE-OBS,INSTRUME,SITEID,TELID,OBSTYPE,BLKUID')
 
+# Metadata that must be present when ingesting a thumbnail
+REQUIRED_THUMBNAIL_METADATA = get_tuple_from_environment('REQUIRED_THUMBNAIL_METADATA', 'frame_basename,size')
```
I think we also need the site_id, instrument_id, and observation_day to be required so we know where to store/retrieve the thumbnails in the daydirs. Actually looking above I think we need to require observation_day for LCO fits files too since its required for that path!

jnation3406 reviewed (approved):
Sorry just noticed this PR! I think we need to add a few fields to required headers, but otherwise looks good.

Commit:
Update required headers for FITS and thumbnails, update tests

Commit:
Update get_filestore_path to be static and take in some frame metadata

This is so that the archive can call this function in the thumbnail model without making an extra join to get all of the header information from the associated frame. Rather, it can provide a set of metadata to simply build the correct file path.

Commit:
Modify get_filestore_path for ThumbnailFile

We keep the same interface for get_filestore_path so that the ingester can utilize it, but add a static method that takes a subset of frame metadata to also generate
a filestore path so that we don't have to pull the entire frame header from another table in the archive. Add thumbnail file and test for ingesting to filestore

Commit:
Fix typo

Commit:
Simplify get_filestore_path. Always assume we have a valid header.

Commit:
Make sure we normalize null values for all thumbnail file metadata, too

Break out the INTEGER_TYPES from the fits file to the settings module

Commit:
Use coveralls parallel build

This is to address the 422 errors we ocassionally get from coveralls

Commit:
Fix to coveralls config. Also fix codacy issues

Commit:
Try coverage fix again.

Commit:
Add debug logging to figure out coveralls issue

Commit:
Convert coverage to xml since the coveralls action can't find coverage in the runner environment.

Commit:
Remove debug logging from coveralls now that it's happy :)

Commit:
Update pyproject.toml, changelog
